packages/astro-embed-baseline-status/BaselineStatus.astro

```diff
@@ -0,0 +1,293 @@
+
+const BASELINE_DEFS = {
+	limited: {
+		title: 'Limited availability',
+		defaultDescription:
+			'This feature is not Baseline because it does not work in some of the most widely-used browsers.',
+	},
+	newly: {
+		title: '',
+		defaultDescription:
+			'This feature works across the latest devices and browser versions. This feature might not work in older devices or browsers.',
+	},
+	widely: {
+		title: 'Widely available',
+		defaultDescription:
+			'This feature is well established and works across many devices and browser versions.',
+	},
+	no_data: {
+		title: 'Unknown availability',
+		defaultDescription:
+			'We currently don’t have browser support information about this feature.',
+	},
+};
+
+/**
+ * Returns feature's `low_date` as mm-yyyy string or empty string if `low_date` is not present.
+ */
+function getBaselineDate(feature: { baseline: { low_date?: string } }): string {
+	return feature.baseline.low_date
+		? new Intl.DateTimeFormat('en-US', {
+				year: 'numeric',
+				month: 'long',
+		  }).format(new Date(feature.baseline.low_date))
+		: '';
+}
+
+/**
+ * Returns Baseline's description.
+ */
+function getDescriptionDate(baseline: StatusLevel, date: string): string {
+	if (baseline === 'newly' && date) {
+		return `Since ${date} this feature works across the latest
+			devices and browser versions. This feature might not work in older
+			devices or browsers.`;
+	} else if (baseline === 'widely' && date) {
+		return `This feature is well established and works across many
+			devices and browser versions. It’s been available across browsers
+			since ${date}`;
+	}
+	return BASELINE_DEFS[baseline].defaultDescription;
+}
+
+const featureData = await safeGet(API_ENDPOINT + Astro.props.id);
+const feature: Feature = featureData?.baseline
+	? (featureData as Feature)
+	: {
+			baseline: {
+				status: 'no_data',
+			},
+			name: Astro.props.id || 'Unknown feature',
+	  };
+
+const baseline = feature.baseline.status || 'no_data';
+
+const title = BASELINE_DEFS[baseline].title;
+const baselineDate = getBaselineDate(feature);
+const description = getDescriptionDate(baseline, baselineDate);
+const year =
+	baseline === 'newly' && baselineDate ? baselineDate.split(' ')[1]! : '';
+const { chrome, edge, firefox, safari } = feature.browser_implementations || {};
+
+function getAriaLabel(
+	title: string,
+	year: string,
+	newlyAvailable: boolean,
+	chrome = 'no',
+	edge = 'no',
+	firefox = 'no',
+	safari = 'no'
+) {
+	if (title === 'Unknown availability') {
+		chrome = edge = firefox = safari = 'unknown';
+	}
+	return `Baseline: ${title}${year ? ` ${year}` : ''}${
+		newlyAvailable ? ` (newly available)` : ''
+	}. Supported in Chrome: ${
+		chrome === 'available' ? 'yes' : chrome
+	}. Supported in Edge: ${
+		edge === 'available' ? 'yes' : edge
+	}. Supported in Firefox: ${
+		firefox === 'available' ? 'yes' : firefox
+	}. Supported in Safari: ${safari === 'available' ? 'yes' : safari}.`;
+}
+---
+
+<baseline-status>
```

Comment:
The custom element names could be removed from all places. If you do need a custom element, it should be namespaced to prevent conflicts, e.g. `<ae-baseline-status>`